### PR TITLE
QWIN-36915 : Mention users in an article

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

**Review by ahamdi — CHANGES_REQUESTED** [review]
Build is failing because we need a UT to cover this fix.

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 1]
```diff
@@ -1185,4 +1179,56 @@ public void unarchiveNews(String newsId) throws Exception {
...
+        if (identity != null) {
+          next = "<p><a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
+                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
+        }
+      } else if (next.contains("@")) {
+        String username = next.split("@")[1];
+        org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
+        if (identity != null) {
+          next = next.split("@")[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
+                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java:1225]: Please try to merge all those conditions in one by directly checking the string after the @ character . Note that we can expected any type of HTML entities or any other characters

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 1 above)
Comment (Rdidmohamedelhadi) [reply services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: done

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**